controllers/components/kserve/kserve_controller.go

```diff
@@ -65,22 +60,20 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 		// changes. The compareHashPredicate ensures that we don't needlessly enqueue
 		// requests if there are no changes that we don't care about.
 		Owns(&templatev1.Template{}, reconciler.WithPredicates(hash.Updated())).
-		// The FeatureTrackers are created slightly differently, and have
-		// ownerRefs set by controllerutil.SetOwnerReference() rather than
-		// controllerutil.SetControllerReference(), which means that the default
-		// eventHandler for Owns won't work, so a slightly modified variant is
-		// added here
-		Owns(&featuresv1.FeatureTracker{}, reconciler.WithEventHandler(
-			handler.EnqueueRequestForOwner(
-				mgr.GetScheme(),
-				mgr.GetRESTMapper(),
-				&componentApi.Kserve{},
-			))).
 		Owns(&networkingv1.NetworkPolicy{}).
 		Owns(&monitoringv1.ServiceMonitor{}).
 		Owns(&admissionregistrationv1.MutatingWebhookConfiguration{}).
 		Owns(&admissionregistrationv1.ValidatingWebhookConfiguration{}).
 		Owns(&appsv1.Deployment{}, reconciler.WithPredicates(resources.NewDeploymentPredicate())).
+
+		// operands - dynamically owned
+		OwnsGVK(gvk.Gateway, reconciler.Dynamic()).
+		OwnsGVK(gvk.EnvoyFilter, reconciler.Dynamic()).
+		OwnsGVK(gvk.KnativeServing, reconciler.Dynamic()).
+		OwnsGVK(gvk.ServiceMeshMember, reconciler.Dynamic()).
+		OwnsGVK(gvk.AuthorizationPolicy, reconciler.Dynamic()).
+		OwnsGVK(gvk.AuthorizationPolicyv1beta1, reconciler.Dynamic()).
```

Comment:
probably those dynamic watching requires a custom predicate that check for serverless being enabled or not  ? similar to https://github.com/opendatahub-io/opendatahub-operator/blob/main/controllers/services/auth/auth_controller.go#L61

Reply:
This is one of the things that I was going to call out in a self-review, that these dynamic ones didn't seem to actually be triggering if the resources changed. Thanks for the suggestion & link, I'll take a look :+1: